packages/opencensus-core/src/metrics/cumulative/derived-cumulative.ts

```diff
@@ -0,0 +1,170 @@
+      void {
+    validateArrayElementsNotNull(
+        validateNotNull(labelValues, 'labelValues'), 'labelValue');
+    validateNotNull(objOrFn, 'obj');
+
+    const hash = util.hashLabelValues(labelValues);
+    if (this.registeredPoints.has(hash)) {
+      throw new Error(
+          'A different time series with the same labels already exists.');
+    }
+    if (this.labelKeysLength !== labelValues.length) {
+      throw new Error('Label Keys and Label Values don\'t have same size');
+    }
+
+    if (objOrFn instanceof Function) {
+      this.extractor = objOrFn;
+    } else if (util.isToValueInterface(objOrFn)) {
+      this.extractor = () => objOrFn.getValue();
+    } else if (util.isLengthAttributeInterface(objOrFn)) {
+      this.extractor = () => objOrFn.length;
+    } else if (util.isLengthMethodInterface(objOrFn)) {
+      this.extractor = () => objOrFn.length();
+    } else if (util.isSizeAttributeInterface(objOrFn)) {
+      this.extractor = () => objOrFn.size;
+    } else if (util.isSizeMethodInterface(objOrFn)) {
+      this.extractor = () => objOrFn.size();
+    } else {
+      throw new Error('Unknown interface/object type');
+    }
+
+    this.registeredPoints.set(
+        hash, {labelValues, extractor: this.extractor, prevValue: 0});
+  }
+
+  /**
+   * Removes the TimeSeries from the cumulative metric, if it is present. i.e.
+   * references to previous Point objects are invalid (not part of the
+   * metric).
+   *
+   * @param {LabelValue[]} labelValues The list of label values.
+   */
+  removeTimeSeries(labelValues: LabelValue[]): void {
+    validateNotNull(labelValues, 'labelValues');
+    this.registeredPoints.delete(util.hashLabelValues(labelValues));
+  }
+
+  /**
+   * Removes all TimeSeries from the cumulative metric. i.e. references to all
+   * previous Point objects are invalid (not part of the metric).
+   */
+  clear(): void {
+    this.registeredPoints.clear();
+  }
+
+  /**
+   * Provides a Metric with one or more TimeSeries.
+   *
+   * @returns {Metric} The Metric, or null if TimeSeries is not present in
+   *     Metric.
+   */
+  getMetric(): Metric|null {
+    if (this.registeredPoints.size === 0) {
+      return null;
+    }
+    const timestamp: Timestamp = getTimestampWithProcessHRTime();
+    return {
+      descriptor: this.metricDescriptor,
+      timeseries: Array.from(
+          this.registeredPoints,
+          ([_, cumulativeEntry]) => {
+            const newValue = cumulativeEntry.extractor();
+            const value = newValue > cumulativeEntry.prevValue ?
+                newValue :
+                cumulativeEntry.prevValue;
+            cumulativeEntry.prevValue = value;
+
+            return {
+              labelValues:
+                  [...cumulativeEntry.labelValues, ...this.constantLabelValues],
+              points: [{value, timestamp}],
+              startTimestamp: this.startTime
+            } as TimeSeries;
```

Comment:
What would you think about moving the type from here to the arrow function signature?

So it would become ` ([_, cumulativeEntry]): TimeSeries => {`. I believe the `as TimeSeries` means that the compiler wouldn't report an error if the fields in the structure don't match the type correctly, whereas specifying it as a return type would.

Reply:
> I believe the as TimeSeries means that the compiler wouldn't report an error if the fields in the structure don't match the type correctly, whereas specifying it as a return type would

Hmm, you are correct. Good catch! Thanks.